pkg/controller/ecs/service/setup.go

```diff
@@ -80,6 +80,11 @@ func postObserve(_ context.Context, cr *svcapitypes.Service, resp *svcsdk.Descri
 		return obs, err
 	}
 
+	if resp.Services[0].DesiredCount != resp.Services[0].RunningCount {
+		obs.ResourceUpToDate = false
+		return obs, nil
+	}
```

Comment:
This does not change the `READY` condition for the resource, but instead it triggers a rereconcilation which is not what we want here.

To change the ready state one needs to use `cr.SetConditions()`.

Reply:
But, don't we want a rereconciliation here? What I'm advovating for is exactly that: we should not consider the resource ready if the `DesiredCount` != `RunningCount`.

I can add a `cr.SetConditions(Unavailable())`, but I followed the logic of the condition just before.

Reply:
No, a reconcilation should only be scheduled explicitly if the (a) the reconcilation itself failed in case of an error or (b) we need a second run to finish the whole reconcilation process.

In this case the resource is just not ready yet. If we would always trigger a rereconcilation we would constantly spam AWS with request and may end up in request limit that would effect other resources as well. It is safe to return with `ResourceUpToDate = true` and set the condition to `Unavailable` or `Creating` and return later at a time that is chosen by the controller runtime.

Reply:
Ok I think I got it. Thanks for the explanation. Already updated the PR accordingly.